collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -297,6 +297,21 @@ private void defaultExport(
           type, help);
       }
 
+      String toSnakeAndLowerCase(String attrName) {
+        if (attrName == null || attrName.isEmpty()) {
+          return attrName;
+        }
+        StringBuilder resultBuilder = new StringBuilder(attrName.substring(0,1).toLowerCase());
+        for (char attrChar : attrName.substring(1).toCharArray()) {
+          if (Character.isUpperCase(attrChar)) {
```

Comment:
This isn't doing exactly the same thing. Consider TCP for which should become tcp but this function would return t_c_p.

Reply:
I'll fix this; I'll add these two tests:
```
TCP -> tcp
testTCPTest -> test_tcptest
```